### DISCUSSION TITLE
Question: how to control 3x2 grid on front page

### DISCUSSION THREAD

**Discussion by epogrebnyak** [thuliteio/doks#368] [Q&A]
There are nice 6 blocks at https://gethyas.com/ front page:

![изображение](https://user-images.githubusercontent.com/9265326/96705438-fb30ba80-139d-11eb-852d-fd850b4eb54f.png)

My question is - are they controlled by html in `index.html`, not by somemarkdown in `contents` folder or config, correct?

https://github.com/h-enk/gethyas.com/blob/8763a6b1b4f05794d9b59e082c4dd357f4ee79c5/layouts/index.html

**Comment by h-enk** [comment]
Yes, you're correct: in the `{{ define "sidebar-prefooter" }}..{{ end }} ` section of `layouts/index.html`.

It would be nicer to separate content from structure, and have the content indeed in `config/_default/params.toml` or in the front matter of `content/_index.md`.

**Comment by epogrebnyak** [comment]
@h-enk, thank you for the quick answer! 

My guess is that frontpage can still be a hard-coded HTML at `layouts/index.html`, but it needs a bit of documentation how is it achieved. 

For me the selling point of the theme is the way how https://causalscience.org/ and https://gethyas.com/ look like, and they arequite different on the front page, a lot of custom HTML. If moving content to `params.toml` or `_index.md` one would need to fix one presentation.

Maybe a new user might need a step-by-step guide for how https://gethyas.com/  is created? Could be a part of its readme file, or a separate blog. Knowing the index page is hardcoded in `layouts/index.html` definetely lowers the cognitive load.

**Comment by h-enk** [comment]
@epogrebnyak, thank you for sharing your thoughts. That helps!

In line with the (original) idea for Hyas, I agree to keep the page setup flexible. So, I will keep it that way, and I will add a guide to the docs explaining how to make use of it.

Hyas is a starter with which you can build a custom theme — like e.g. the documentation theme used by gethyas.com, or the conference theme used by causalscience.org. Would you be interested in other single purpose (but flexible) custom themes build on top of Hyas? And if, what type of themes would that be — e.g. a blog theme, or an alternative to the Academic theme?

**Comment by epogrebnyak** [comment]
Academic is a single most popular theme for Hugo, I think, it has a lot of fucntionalities to embed twitter, youtube, keep the people directory and (huge advantage) multi-language versions , but I think visually it is not a modern one. I have a personal site built with Academic, https://epogrebnyak.github.io/ but I'm not too happy how it looks. 

I'm actually surprised Hyas is not a popular base theme yet - it is exceptionally clean. 

If it helps I think the theme can work in several "presets" or usecases, appealing for different audiences:
- documentation theme 
- landing page (with custom index.html)
- other (?)

Can elaborate on what peope look for in either case. Chasing Academic itself maybe not a great idea, it has too rich of functionality.

An important gap in documentation is why exactly people need `yarn` for Hugo theme, important for people not familiar with  JS ecosystem.

Will write more later.

**Comment by epogrebnyak** [comment]
Extra thouhgts (very subjective):

1. There are three key usecases for static site generators - documentation, landing page, blog and tentative combinations of the above.
2. There are few leading themes per each static site generator:
  - mkdocs - mkdocs-material
  - hugo - Academic and several blog themes
  - sphinx - read the docs theme
3. In Hugo in specific there is a lot of effort to provide many themes as opposed to focus on few well-tested themes. A lot of themes just survive from Wordpress and Jekyll, but porting does not work too well - need maintenance, etc. 
4. I know less about JavaScript-based site generators maybe they have their own trends.

`hyas` is a great candidate for a cool documentation, landing and Academic-style theme, maybe also a blog (but for blogs there are already too many). As a suggestion - maybe `hyas` can support three following templates in separate repos:

- `hyas-docs` - giving acces to [this layout](https://gethyas.com/docs/prologue/introduction) out of the box, with functionality similar to https://squidfunk.github.io/mkdocs-material
- `hyas-landing` - describing how to achieve a look similar to https://gethyas.com.  Academic theme can be a prototype for further development of `hyas-landing`.
- `hyas-blog` - similar to https://gethyas.com/blog/.

On main page of documentation it would be nice if these tree usecases wherer highlighted - a user would be happy to have out-of the box presets for docs, blog, or landing. For me as an intermediate user saying - here is a 'starter' theme sounds like a risk - oh no, I do not want to do any emngineering, I want to get using the theme right away (as mkdocs-marterial beautifully enables to do).

As mentioned above, need to give more instruction about Javascript and `yarn` - a combination of needing to install both go and js may be a hard to task for many users, eg python users. When there is an explaination why these tools where used people would feel more comfortable using `hyas`.

Hope it helps.

**Comment by h-enk** [comment]
Thank you for your elaborate answer — appreciated! It really helps me to see things from a different perspective.

Converging things, this is how I see it now.

Hyas serves three types of users:
- beginner
- intermediate
- expert

Hyas — the starter (example: https://hyas.netlify.app/) is for expert users who want to make their own theme (note: https://gethyas.com/ is about https://hyas.netlify.app/).

Ready made themes build on top of Hyas are for beginners and intermediate users mainly. For example:
- a documentation theme (like https://gethyas.com/ — the whole site that is)
- a blog theme (like https://henkverlinde.com/)
- a landing page theme (like https://getvalidate.com/)

I will definitely release the https://gethyas.com/ website as a separate documentation theme (because that's where the interest is). I keep a blog theme and landing page theme in mind (I'm forgettting about Academic for now). 

I will add a themes section to https://gethyas.com/, so you can easily pick the theme for your usecase.

I will explain /elaborate more in the docs (https://gethyas.com/docs) on a/o:
- starter versus theme
- update installation instructions
- why Node/npm: needed for installing dependencies (styles + scripts) and running additional commands — in short
- why yarn: a better npm — in short
- how to set up landing pages (like the homepage)
- how to customize look & feel (think colors, fonts et cetera)
- how to add content (think pages)

Going forward ;)

**Comment by epogrebnyak** [comment]
The division of user perspectives is great! Most projects have just one entry-point and it is scary (lots of things to learn, many risks, too long before doing soemthign useful). Couple follow-ups:

- _starter_ is not a great name for intermediate level project template, maybe _canvas_. Beginners would think the starter is for them and it is not.
- docs is rightly the main/central usecase, next most needed is docs + blog is next in line, at least in my projects I want to add somethign less formal than a site page, maybe temporary, but in mkdocs or sphinx you end up mimicing the blog inside docs, not too great looking
- hyas itself is a theme for Hugo, when saying theme for hyas, you position hyas as a static site generator on its own (rightfully - it has own cli and packaging), but keeping a word theme has implication. template/preset could be alternatives 
- may reconsider a tagline _Hyas is a Hugo starter helping you build modern websites that are secure, fast, and SEO-ready — by default._, to me unclear is starter here? Hyas is a set of Hugo themes... Hyas is supplement to Hugo... is a collection of themes... (may leave starter here, maybe a better term emerges later)

Thanks you for all the work and looking forward to try hyas docs or docs+blog theme is any of my projects. Please let me know if I can help if hyas development or testing.

**Comment by epogrebnyak** [comment]
Small usecase: I cloned https://github.com/h-enk/gethyas.com to try to change appearance of index page to be pointing to 
https://gethyas.com/docs/prologue/introduction/ and I could not even guess where to start doing that. Really need `hyas-docs` example.

**Comment by epogrebnyak** [comment]
For noobs - maybe type of page template could be a flag for init command? Like `hyas create --docs`?

**Comment by epogrebnyak** [comment]
Closing as initial question is answered
